GenderPayGap.WebUI/Models/ManageOrganisations/ManageOrganisationViewModel.cs

```diff
@@ -128,64 +131,63 @@ private bool OrganisationIsRequiredToSubmit()
                    && !Global.ReportingStartYearsToExcludeFromLateFlagEnforcement.Contains(GetAccountingDate().Year);
         }
 
-        private ReportStatus GetReportStatus()
+        private ReportTag GetReportStatus()
         {
             Return returnForYear = organisation.GetReturn(ReportingYear);
-            bool reportIsNotSubmitted = returnForYear == null;
-
-            if (OrganisationIsRequiredToSubmit() && reportIsNotSubmitted && !DeadlineHasPassed())
-            {
-                return ReportStatus.Due;
-            }
+            bool reportIsSubmitted = returnForYear != null;
+            
+            return reportIsSubmitted ? GetSubmittedReportStatus(returnForYear) : GetNotSubmittedReportStatus();
+        }
 
-            if (OrganisationIsRequiredToSubmit() && reportIsNotSubmitted && DeadlineHasPassed())
+        private ReportTag GetSubmittedReportStatus(Return returnForYear)
+        {
+            bool returnIsRequired = returnForYear.IsRequired();
+            
+            if (returnIsRequired && !returnForYear.IsLateSubmission || !returnIsRequired && returnForYear.IsSubmitted())

```

Comment:
Why do we need to check if the return is submitted? We already do that before calling the method.
I'd revert the order to simplify this method. Something like this:
```
if (returnIsRequired && isLateSubmission) {
   return SubmittedLate;
}
return Submitted;
```